include/swift/AST/DiagnosticsSema.def

```diff
@@ -1071,6 +1071,9 @@ ERROR(optional_base_not_unwrapped,none,
 NOTE(optional_base_chain,none,
      "chain the optional using '?' to access member %0 only for non-'nil' "
      "base values", (DeclNameRef))
+NOTE(optional_base_remove_optional_for_keypath_root, none,
+    "remove optional '?' form key path root base type %0", (Type))
```

Comment:
```suggestion
    "remove optional '?' from key path root base type %0", (Type))
```

Reply:
Also, “change Foo? to Foo” might be a simpler message.

Reply:
Nice catch, thanks!

> Also, “change key path base type from Foo? to Foo” maybe?

 I think this is the implicit message already provided by the fixItReplace we have "String?" -> "String" so I'm not sure it would add much adding this in the note. And since what we want to indicate is remove optionality, I think the note should say something along these lines :)